test/spmd/tensor/test_tensor_ops.py

```diff
@@ -84,6 +84,40 @@ def test_op_out_variant(self):
         self.assertTrue(res is dist_tensor_out)
         self.assertEqual(dist_tensor_out.to_local(), expected_dt.to_local())
 
+    @with_comms
+    def test_empty_like(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        shard_spec = [Shard(0)]
+
+        input_tensor = torch.randn(4, 8, requires_grad=True)
+        dist_tensor = DTensor.from_local(input_tensor, device_mesh, shard_spec)
+        empty_like_dt = torch.empty_like(dist_tensor)
+        # empty is not deterministic, so we only check that the shard propagation worked
```

Comment:
Is empty not deterministic even if we set the seed on each rank to be the same?

Reply:
I believe it simply doesn't initialize the memory so it's probably going to be just trash.